src/core/core/task.cpp

```diff
@@ -366,18 +367,18 @@ void task::enqueue()
 void task::enqueue(task_worker_pool* pool)
 {
     this->add_ref(); // released in exec_internal (even when cancelled)
-    
+
+    if (spec().type == TASK_TYPE_COMPUTE)
+    {
```

Comment:
yes, my expected, especially for simulator network, where it uses task delay to simulate rpc delay, we need more info.